libraries/AP_Telemetry/MQTT_Mavlink.cpp

```diff
@@ -0,0 +1,275 @@
+            msg,
+            0,0,MAV_CMD_NAV_TAKEOFF,0,
+            0.0,0.0,hnbpa,0.0,0.0,0.0,takeoff_alt);
+        ret = 1;
+    } else if (strncmp(cmd, "mode land", 9) == 0) {
+        //mode land
+        char buf[30] = {0};
+        float hnbpa = 1.0; // param 3 horizontal navigation by pilot acceptable
+        msg->msgid = MAVLINK_MSG_ID_COMMAND_LONG;
+        mavlink_msg_command_long_pack_chan(
+            0,0,0,
+            msg,
+            0,0,MAV_CMD_NAV_LAND,0,
+            0.0,0.0,hnbpa,0.0,0.0,0.0,0.0);
+        msg->msgid = MAVLINK_MSG_ID_SET_MODE;
+        msg->len = 6;
+        _mav_put_uint32_t(buf, 0,9);
+        _mav_put_uint8_t(buf, 4,1);
+        _mav_put_uint8_t(buf, 5,1);
+        memcpy(_MAV_PAYLOAD_NON_CONST(msg), buf, 30);// 30 is about
+        memset(msg, 0, sizeof(mavlink_message_t));
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Landing receive", topic);
+        ret = 1;
+    } else if ((strncmp(cmd, "mode alt_hold", 13) == 0) || (strncmp(cmd, "mode althold", 12) == 0)) {
+        //mode alt hold
+        char buf[30] = {0};
+        memset(msg, 0, sizeof(mavlink_message_t));
+        msg->msgid = MAVLINK_MSG_ID_SET_MODE;
+        msg->len = 6;
+        _mav_put_uint32_t(buf, 0,2);
+        _mav_put_uint8_t(buf, 4,1);
+        _mav_put_uint8_t(buf, 5,1);
+        memcpy(_MAV_PAYLOAD_NON_CONST(msg), buf, 30);// 30 is about
+        memset(msg, 0, sizeof(mavlink_message_t));
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Alt old recevied", topic);
+        ret = 1;
+    } else if (strncmp(cmd, "mode loiter", 11) == 0) {
+        //mode loiter
+        char buf[30]= {0};
+        memset(msg, 0, sizeof(mavlink_message_t));
+        msg->msgid = MAVLINK_MSG_ID_SET_MODE;
+        msg->len = 6;
+        _mav_put_uint32_t(buf, 0,5);
+        _mav_put_uint8_t(buf, 4,1);
+        _mav_put_uint8_t(buf, 5,1);
+        memcpy(_MAV_PAYLOAD_NON_CONST(msg), buf, 30);// 30 is about
+        memset(msg, 0, sizeof(mavlink_message_t));
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Loiter mode received", topic);
+        ret = 1;
+    } else if (strncmp(cmd, "mode guided", 11) == 0) {
+        //mode guided
+        char buf[30] = {0};
+        memset(msg, 0, sizeof(mavlink_message_t));
+        msg->msgid = MAVLINK_MSG_ID_SET_MODE;
+        msg->len = 6;
+        _mav_put_uint32_t(buf, 0,4);
+        _mav_put_uint8_t(buf, 4,1);
+        _mav_put_uint8_t(buf, 5,1);
+        memcpy(_MAV_PAYLOAD_NON_CONST(msg), buf, 30);// 30 is about
+        memset(msg, 0, sizeof(mavlink_message_t));
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Guided mode received", topic);
+        ret = 1;
+    } else if (strncmp(cmd, "mode circle", 11) == 0) {
+        //mode circle
```

Comment:
this approach to coding things manually for specific cases just isn't going to scale.